examples/AdvancedOperations/AddDynamicSearchAds.php

```diff
@@ -0,0 +1,347 @@
+        // into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments([
+            ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT
+        ]);
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId
+    ) {
+        $budgetResourceName = self::createCampaignBudget($googleAdsClient, $customerId);
+        $campaignResourceName = self::createCampaign(
+            $googleAdsClient,
+            $customerId,
+            $budgetResourceName
+        );
+        $adGroupResourceName = self::createAdGroup(
+            $googleAdsClient,
+            $customerId,
+            $campaignResourceName
+        );
+        self::createExpandedDSA($googleAdsClient, $customerId, $adGroupResourceName);
+        self::createWebPageCriteria($googleAdsClient, $customerId, $adGroupResourceName);
+    }
+
+    /**
+     * Creates a campaign budget.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client.
+     * @param int $customerId the client customer ID.
+     * @return string the campaign budget resource name.
+     */
+    private static function createCampaignBudget(GoogleAdsClient $googleAdsClient, $customerId)
```

Comment:
This is inconsistent with your last method, where you use a string as type hint for $customerId

Reply:
Done.